src/object/object_primitive.c

```diff
@@ -12015,55 +11989,20 @@ mr_cmpval_char (DB_VALUE * value1, DB_VALUE * value2, int do_coercion, int total
   size1 = db_get_string_size (value1);
   size2 = db_get_string_size (value2);
 
-  /*
-   * do_coercion = 2:
-   * from btree compare key
-   * we need to process the ignore trailing space
-   */
-  if (do_coercion == 2)
-    {
-      ti = true;
-    }
-  /*
-   * do_coercion = 3:
-   * from eliminate_duplicated_keys and scan_key_compre
-   * we need to process enforcing no-ignore-trailing space.
-   */
-  else if (do_coercion == 3)
-    {
-      ti = false;
-    }
-  else if (!ignore_trailing_space &&
-	   (type1 == DB_TYPE_STRING || type1 == DB_TYPE_VARNCHAR || type2 == DB_TYPE_STRING
-	    || type2 == DB_TYPE_VARNCHAR))
+  if (!ignore_trailing_space)
     {
-      int i;
-
-      if (type1 == DB_TYPE_CHAR || type1 == DB_TYPE_NCHAR)
+      if (do_coercion == 2)
```

Comment:
```suggestion
      if (do_coercion != 2 && 
          (!TP_IS_FIXED_LEN_CHAR_TYPE (type1) || !TP_IS_FIXED_LEN_CHAR_TYPE (type2)))
        {
          ti = false;
        }
```

Reply:
I understand your comment, but I keep the code and comment for readability.